### PR TITLE
EVG-20089 Add useEffectDebugger hook

### PR DESCRIPTION
**Pull request by khelif96**
EVG-20089

### Description
This is just copying over the useEffectDebugger util hook from parsley over to spruce. 
Decided to split this off of EVG-18780

### CODE REVIEW

File: src/test_utils/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * @param initialValue
+ * @returns - the previous value
+ */
+const usePrevious = <T>(value: T, initialValue: T) => {
+  const ref = useRef(initialValue);
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
```
Comment (minnakt) [inline src/test_utils/hooks.ts:15]: Do we need to redeclare `usePrevious` if we have an existing one in the hooks folder? :0

**Review by minnakt — APPROVED** [review]
yayayayay

File: src/test_utils/hooks.ts (hunk 1 above)
Comment (khelif96) [reply src/test_utils/hooks.ts]: Great catch!

Comment (khelif96) [reply src/test_utils/hooks.ts:15]: Actually, nvm they are slightly different. One of them declares an initialValue Param.

**Pull request merged**